apis/compute/v1alpha1/substitutions_types.go

```diff
@@ -0,0 +1,14 @@
+package v1alpha1
+
+// Substitution defines the substitution configuration
+type Substitution struct {
```

Comment:
@mwaionos WDYT?

Reply:
Hey Filip the `value` for `type` is generated by `substitution`, that's why it's not part of the incoming spec. In common `keys` are unique, whereas `placeholderName` feels more like it could be reused.

The substitution parses the incoming document for `key` and replaces it's value by the generated outcome of `type()`.

That's why i think that we should stick with a more common, clear naming convention.

Reply:
> Maybe we could give also an example of a cloud-config e.g. (not 100% if this is how cloud-config field value really looks like. I took as reference the values from pkg/ccpatch/substitutions_test.go

The cloudconfig is up to the user, in our case we have our own cloud-config implementation. But the most users will stick with cloudinit: https://cloudinit.readthedocs.io/en/latest/reference/examples.html

Reply:
We cam to a conclusion and we will add some coments e.g.
```
// Substitution defines the substitution configuration
// Example:
// substitutions:
//   - type: ipv4Address
//     key: $keyToReplace
//     unique: true
//     options:
//     cidr: "10.0.0.0/24"
type Substitution struct {
	// +kubebuilder:validation:Required
	// +immutable
	// +kubebuilder:validation:Enum=ipv4Address;ipv6Address
	// The Type of the handler that will be used by this substitution. The handler will
	// be responsible for computing the value to be replaced by the substitution key
	Type string `json:"type" yaml:"type"`
	// +kubebuilder:validation:Required
	// +immutable
	Key string `json:"key" yaml:"key"`
	// Unique means that the value is unique across multiple ServerSets
	// +immutable
	Unique bool `json:"unique" yaml:"unique"`
	// The options for the handler
	Options map[string]string `json:"options" yaml:"options"`
}
```

Reply:
See commit: https://github.com/ionos-cloud/crossplane-provider-ionoscloud/pull/235/commits/8cad684495a089bb4a9f3968839ae8d4f9fea2dc